docs/workflow/ci/failure-analysis.md

```diff
@@ -0,0 +1,118 @@
+
+An issue that's not been reported before it will look like this in the `Build Analysis` check tab:
+
+![failed test](failed-test.png)
+
+You can use the console log, any potential attached dumps in the artifacts section, or any other piece of information printed to help you decide if it's a regression caused by the change. Similarly, for runtime tests we will try to print the crashing stacks to aid in the investigation.
+
+If you have considered all the diagnostic artifacts and determined the failure is definitely not caused by changes in your PR, please do this:
+
+1. Identify a string from the logs that uniquely identifies the issue at hand. A good example of this the string `The system cannot open the device or file specified. : 'NuGet-Migrations'` for issue https://github.com/dotnet/runtime/issues/80619.
+2. On the test failure in the tab you can select `Report repository issue`. This will prepopulate an issue with the appropriate tags and with a body similar to:
+    ````
+    Build Information
+    Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=242380
+    Build error leg or test failing: Build / linux-arm64 Release AllSubsets_Mono_Minijit_RuntimeTests minijit / Build Tests
+    Pull request: https://github.com/dotnet/runtime/pull/84716
+    <!-- Error message template  -->
+    ## Error Message
+    Fill the error message using [known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-section).
+
+    ```json
+    {
+        "ErrorMessage": "",
+        "BuildRetry": false,
+        "ErrorPattern": "",
+        "ExcludeConsoleLog": false
+    }
+    ```
+    ````
+    It already contains most information necessary. *The most important piece of information for you to fill out is the json blob*.
+
+    - You can add the string you identified as uniquely identifying the issue into the `ErrorMessage` field. In case you need to use a regex, use the `ErrorPattern` field instead (this is a limitted, single-line, non-backtracking regex described [here](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#regex-matching)). This regex needs to be appropriately escaped - check the [arcade known issues](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#filling-out-known-issues-json-blob) documentation for a good guide on proper regex and JSON escaping.
+    - The field `ExcludeConsoleLog` should be `true` as much as possible. Setting it to false will mean all failures appear in the same log and since you didn't exclude the overall logs, all failures in the collection will get attributed to this one's issue bucket and errors will not be investigated properly. Due to limitations in Known Issues around rate limiting and xUnit resiliency, setting `ExcludeConsoleLog=false` is necessary in two scenarios:
```

Comment:
```suggestion
    - The field `ExcludeConsoleLog` describes if the execution logs should be considered on top of the individual test results. **For most cases, this should be set to `true` as the failure will happen within a single test**. Setting it to `false` will mean all failures within an xUnit set of tests will also get attributed to this particular error, since there's one log describing all the problems. Due to limitations in Known Issues around rate limiting and xUnit resiliency, setting `ExcludeConsoleLog=false` is necessary in two scenarios:
```